### PR TITLE
Fix: subdirectory deployment

### LINKED ISSUE #2128: [Bug] base and startUrl invalid
### Description

base and startUrl invalid setting 

### Steps to reproduce

After setting the base: http://host.local/homepage
After that, we can only pass through http://host.local/ Unable to access homepage

### homepage version

v 0.7.2

### Installation method

Docker

### Configuration

```yaml
base: http://host.local/homepage
startUrl: http://host.local/homepage
```

### Container Logs

_No response_

### Browser Logs

_No response_

### Troubleshooting

After setting the base: http://host.local/homepage
After that, we can only pass through http://host.local/ Unable to access homepage

### Other

_No response_

### Before submitting, I have made sure to

- [X] Check [the documentation](https://gethomepage.dev/)
- [X] Follow [the troubleshooting guide](https://gethomepage.dev/latest/more/troubleshooting/) (please include output above if applicable).
- [X] Search [existing issues](https://github.com/gethomepage/homepage/search?q=&type=issues) and [discussions](https://github.com/gethomepage/homepage/search?q=&type=discussions).

### PR DESCRIPTION
**Pull request by alfred-nice**
Modify the absolute path to a relative path, and support the subdirectory deployment of nginx. #2128
#539

## Proposed change

close #2128 #539
supprt nginx subdirectory.
image

## Type of change

Support the nginx subdirectory.

## Checklist:

- README.md. Added how to use nginx in readme file.
- next.config.js. static files change to realtive path.
- src/pages/index.jsx. mod /api/xxx to ./api/xxx. 
- others.... format the code.

`pnpm dev`:
image

`pnpm build`:
image

## result:

static file:
image
/api -> ./api:
image

### CODE REVIEW

**Comment by benphelps** [comment]
I think the correct solution is to simply omit the directory prefix all together.

For example:

```js
const { data, error, isLoading } = useSWR(`./api/config/${ path }`, fetcher)
```

Should really be:

```js
const { data, error, isLoading } = useSWR(`api/config/${ path }`, fetcher)
```

**Comment by benphelps** [comment]
Please also make sure you have minimal linter changes included in the PR, as it makes reviewing this harder than it needs to be (and it's touching a lot of files).

File: README.md [hunk 1]
```diff
@@ -139,8 +139,22 @@ Start the development server:
...
+        }
+```
+
+Using subdirectory, don't ignore the last **/**, visit http://localhost/homepage/:
+```
+location /homepage/ {
+            proxy_pass $homepage_upstream/;
+        }
+```
+
```
Comment (shamoon) [inline README.md:157]: This doesn't belong in the readme

```suggestion
```

**Review by shamoon — CHANGES_REQUESTED** [review]
Please address Ben's comment re. relative path and remove the whitespace changes from the PR

File: README.md (hunk 1 above)
Comment (alfred-nice) [reply README.md:157]: thanks for your suggestions, I will revise it as soon as possible. ^~^

**Pull request merged**